content/tutorials/reactjs-realtime-commenting.textile

```diff
@@ -0,0 +1,344 @@
+          </div>
+        </div>
+      </section>
+    );
+  }
+}
+
+export default App;
+```
+
+This contains the components we created earlier. We define a @comments@ state which is an array of comments. It defaults to an empty array. This will be updated as we add comment through the comment form. Also, we pass the @comments@ state as props to the Comments component. This is how the Comments component get the comments it renders.
+
+h2. Step 13 - Subscribing to Comments
+
+We have seen how we can publish newly added comments. Now, we need to subscribe to those comments being published. We’ll do this within React lifecycle @componentDidMount()@ hook. Still within @components/App.js@, add the code below just before the @render()@ function:
+
+```[javascript]
+componentDidMount() {
+  // Subscribe to comments
+  const channel = Ably.channels.get('comments');
+
+  channel.subscribe('add_comment', (comment) => {
+    this.setState((prevState) => {
+      return {
+        comments: prevState.comments.concat({
+          name: comment.data.name,
+          comment: comment.data.comment
+        })
+      };
+    });
+  });
+}
+```
+
+The @componentDidMount()@ hook runs after the component output has been rendered to the DOM. It is a good place to subscribe to comments. We subscribe to the @comments@ channel and listen for the @add_comment@ event. We then update the @comments@ state with the comment published from the channel we are subscribed to.
+
+<a href="/images/tutorials/reactjs-realtime-commenting-final-output.png" target="_blank">
+  <img src="/images/tutorials/reactjs-realtime-commenting-final-output.png" style="width: 100%" alt="Final output screenshot">
+</a>
+
+That’s it. We now have a realtime commenting system. To test it out, start the app:
+
+```[sh]
+npm start
+```
+
+Then open "http://localhost:3000":http://localhost:3000 in two different browser tabs. Add a comment in one of the opened tabs and watch the other tab update with the comment in realtime.
+
+h2. Download tutorial source code
+
+The complete source code for each step of "this tutorial is available on Github":https://github.com/ably/tutorials/commits/reactjs-realtime-commenting.
+
+We recommend that you clone the repo locally:
+
+```[sh]
+git clone https://github.com/ably/tutorials.git
+```
+
+Checkout the tutorial branch:
+
+```[sh]
+git checkout reactjs-realtime-commenting
+```
+
+Install NPM dependencies. Be sure to switch into project's directory and then run this command in your terminal:
+
+```[sh]
+npm install
+```
+
+And then run the app locally by adding your "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys to @src/ably.js@ and run
+
+```[sh]
+npm start
+```
+
+to start the app.
+
+h2. Next steps
+
+1. If you would like to find out more about how channels, publishing and subscribing works, see the "Realtime channels & messages documentation":https://www.ably.io/documentation/realtime/channels-messages
```

Comment:
Perhaps add in a link to the history tutorial and history documentation as well, see https://www.ably.io/tutorials/history#lang-javascript